src/test/scala/za/co/absa/abris/avro/subject/SubjectNameStrategyAdapterFactorySpec.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2018 ABSA Group Limited
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package za.co.absa.abris.avro.subject
+
+import io.confluent.kafka.serializers.subject.{RecordNameStrategy, TopicNameStrategy, TopicRecordNameStrategy}
+import org.scalatest.FlatSpec
+import za.co.absa.abris.avro.read.confluent.SchemaManager.SchemaStorageNamingStrategies._
+
+class SubjectNameStrategyAdapterFactorySpec extends FlatSpec {
+
+  behavior of SubjectNameStrategyAdapterFactory.getClass.getName
```

Comment:
in `SubjectNameStrategyAdapterSpec` you have `behavior of classOf[SubjectNameStrategyAdapter].getName`, you're using `.getClass` here, you can do either, but keep it consistent. (I think `classOf` is the scala way)

Reply:
Yep, that seems to be the Scala way. However, SubjectNameStrategyAdapterFactory is an object, whereas SubjectNameStrategyAdapter is a class. But dude, that is the proper definition of a keen eye!

Reply:
Wow, didn't realize you can't call `classOf[object]`, that's interesting!